api/types/second_factor.go

```diff
@@ -25,30 +25,13 @@ import (
 	"github.com/gravitational/teleport/api/constants"
 )
 
-// legacySecondFactorFromSecondFactors returns a suitable legacy second factor for the given list of second factors.
-func legacySecondFactorFromSecondFactors(secondFactors []SecondFactorType) constants.SecondFactorType {
-	hasOTP := slices.Contains(secondFactors, SecondFactorType_SECOND_FACTOR_TYPE_OTP)
-	hasWebAuthn := slices.Contains(secondFactors, SecondFactorType_SECOND_FACTOR_TYPE_WEBAUTHN)
-
-	switch {
-	case hasOTP && hasWebAuthn:
-		return constants.SecondFactorOn
-	case hasWebAuthn:
-		return constants.SecondFactorWebauthn
-	case hasOTP:
-		return constants.SecondFactorOTP
-	default:
-		return constants.SecondFactorOff
-	}
-}
-
 // secondFactorsFromLegacySecondFactor returns the list of SecondFactorTypes supported by the given second factor type.
 func secondFactorsFromLegacySecondFactor(sf constants.SecondFactorType) []SecondFactorType {
 	switch sf {
 	case constants.SecondFactorOff:
 		return nil
 	case constants.SecondFactorOptional, constants.SecondFactorOn:
-		return []SecondFactorType{SecondFactorType_SECOND_FACTOR_TYPE_WEBAUTHN, SecondFactorType_SECOND_FACTOR_TYPE_OTP}
+		return []SecondFactorType{SecondFactorType_SECOND_FACTOR_TYPE_OTP, SecondFactorType_SECOND_FACTOR_TYPE_WEBAUTHN}
```

Comment:
Should we put Webauthn first, to reflect the preference for the stronger second factor?

(Or are not we considering the order of second_factors?)

Reply:
Yeah, we're not considering the order of second_factors, We will prefer WebAuthn > SSO > OTP